cli/src/cmd/forge/install.rs

```diff
@@ -204,32 +218,130 @@ fn install_as_submodule(dep: &Dependency, libs: &Path, no_commit: bool) -> eyre:
+    if dep.tag.is_none() {
+        return Ok(String::new())
+    }
+    let tag = match_tag(dep.tag.as_ref().unwrap(), libs, target_dir)?;
+    let url = dep.url.as_ref().unwrap();
+
+    let args = if recurse {
+        vec!["checkout", "--recurse-submodules", &tag]
     } else {
-        format!("forge install: {target_dir}")
+        vec!["checkout", &tag]
     };
+    let output = Command::new("git")
+        .args(args)
+        .current_dir(&libs.join(&target_dir))
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .output()?;
+    let stderr = String::from_utf8_lossy(&output.stderr);
+    if !&output.status.success() {
+        // remove dependency on failed checkout
+        std::fs::remove_dir_all(libs.join(&target_dir))?;
 
-    if !no_commit {
-        Command::new("git")
-            .args(&["commit", "-m", &message])
-            .current_dir(&libs)
-            .stdout(Stdio::piped())
-            .spawn()?
-            .wait()?;
+        if stderr.contains(
+            format!("error: pathspec '{tag}' did not match any file(s) known to git").as_str(),
+        ) {
+            eyre::bail!("Tag: \"{}\" not found for repo: \"{}\"!", tag, url)
+        } else {
+            eyre::bail!("{}", stderr.trim())
+        }
     }
 
-    Ok(())
+    Ok(tag)
+}
+
+/// disambiguate tag if it is a version tag
+fn match_tag(tag: &String, libs: &Path, target_dir: &str) -> eyre::Result<String> {
+    // only try to match if it looks like a version tag
+    if !DEPENDENCY_VERSION_TAG_REGEX.is_match(tag) {
+        return Ok(tag.into())
+    }
+
+    // find matching tags
+    let trimmed_tag = tag.trim_start_matches('v').to_string();
+    let output = Command::new("git")
+        .args(&["tag"])
+        .current_dir(&libs.join(&target_dir))
+        .stdout(Stdio::piped())
+        .output()?;
+    let output = String::from_utf8_lossy(&output.stdout);
+    let candidates: Vec<String> = output
+        .trim()
+        .split('\n')
+        .filter(|x| x.trim_start_matches('v').starts_with(&trimmed_tag))
+        .map(|x| x.to_string())
+        .rev()
+        .collect();
+
+    // no match found, fall back to the user-provided tag
+    if candidates.is_empty() {
+        return Ok(tag.into())
+    }
+
+    // have exact match
+    for candidate in candidates.iter() {
+        if candidate == tag {
+            return Ok(tag.into())
+        }
+    }
+
+    // no exact match
+    if candidates.len() == 1 {
+        // only one candidate, ask whether user wants to accept or not
+        let matched_tag = candidates[0].clone();
+        print!(
+            "Found a similar version tag: {}, do you want to use this insead? ([y]/n)",
+            matched_tag
+        );
+        stdout().flush()?;
+        let mut input = String::new();
+        stdin().read_line(&mut input)?;
+        input = input.trim().to_lowercase();
+        if input.is_empty() || input == "y" || input == "yes" {
```

Comment:
let's return here and get rid of the else branch this way

Reply:
good suggestion, will do that insead